src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -110,18 +103,12 @@ def get_user_agent(self):
         """
 
         def parts():
-            # Be careful not to use "self.user_agent" as it will recursively call get_user_agent in __getattr__
-            agent = self.get(self._USER_AGENT, '')
-            yield agent
-            promptflow_agent = f"promptflow/{VERSION}"
-            yield promptflow_agent if promptflow_agent not in agent else ''
-            user_agent = os.environ.get(USER_AGENT, '').strip()
-            yield user_agent if user_agent not in agent else ''
-            pf_user_agent = os.environ.get(PF_USER_AGENT, '').strip()
-            yield pf_user_agent if pf_user_agent not in agent else ''
+            yield f"promptflow/{VERSION}"
+            if "user_agent" in self:
+                yield self.get("user_agent")
 
         # strip to avoid leading or trailing spaces, which may cause error when sending request
```

Comment:
The comment mentions stripping spaces, but the code doesn't include `strip()`. Please either add `strip()` to match the comment or revise the comment to reflect the current logic.